lib/nerves/artifact/providers/docker.ex

```diff
@@ -124,7 +124,7 @@ defmodule Nerves.Artifact.Providers.Docker do
     ]
 
     mounts = Enum.join(mounts(pkg), " ")
-    cmd = "docker run --rm -it -w #{@working_dir} #{mounts} #{image}"
+    cmd = "docker run --rm -it -w #{@working_dir} #{mounts} #{ssh_agent()} #{image}"
```

Comment:
When you `#{ssh_agent()}` and the list interpolated, I think the output will be "-v$SSH_AUTH_SOCK:/ssh-agent-eSSH_AUTH_SOCK=/ssh-agent". Could that be an issue?